addon/-private/system/model/internal-model.js

```diff
@@ -899,8 +899,12 @@ export default class InternalModel {
         }
       }
     });
-    Object.keys(this._implicitRelationships).forEach((key) => {
-      let rel = this._implicitRelationships[key];
+
+    let implicitRelationships = this._implicitRelationships;
+    this.__implicitRelationships = null;
```

Comment:
This feels a little insider baseball to me but I don't have a better suggestion.